lib/event_sourcery/event_store/memory.rb

```diff
@@ -11,37 +11,34 @@ def initialize(events = [], event_builder: EventSourcery.config.event_builder)
       def sink(event_or_events, expected_version: nil)
         events = Array(event_or_events)
         ensure_one_aggregate(events)
+
         if expected_version && version_for(events.first.aggregate_id) != expected_version
           raise ConcurrencyError
         end
+
         events.each do |event|
-          id = @events.size + 1
-          serialized_body = EventBodySerializer.serialize(event.body)
           @events << @event_builder.build(
-            id: id,
+            id: @events.size + 1,
             aggregate_id: event.aggregate_id,
             type: event.type,
             version: next_version(event.aggregate_id),
-            body: serialized_body,
+            body: EventBodySerializer.serialize(event.body),
             created_at: event.created_at || Time.now.utc,
             uuid: event.uuid
           )
         end
+
         true
       end
 
       def get_next_from(id, event_types: nil, limit: 1000)
-        events = @events.select { |event| event.id >= id }
-        if event_types
-          events = events.select { |event| event_types.include?(event.type) }
-        end
-        events.first(limit)
+        events = event_types.nil? ? @events : @events.select { |e| event_types.include?(e.type) }
```

Comment:
At first I extracted this line into a method but then it had a bit strange behaviour and a bit strange name IMO.